src/book/04-guides/01-nubots/02-camera-calibration.mdx

```diff
@@ -0,0 +1,74 @@
+---
+section: Guides
+chapter: NUbots
+title: Camera Calibration
+description: Learn how to calibrate cameras.
+slug: /guides/nubots/camera-calibration
+---
+
+Camera calibration is done to determine optimal parameters for the cameras, such as focal length and the transform from one camera to the other. More on camera calibration can be found on [the vision page](/system/nubots/vision).
+
+The calibration is done in two steps. The first is creating the nbs recording and the second is using that nbs recording to find the camera parameters.
+
+1. Change `cameratest.role` to include `support::logging::DataLogging`
+
+   ```bash
+   nuclear_role(
+     # FileWatcher, ConsoleLogHandler and Signal Catcher Must Go First. KinematicsConfiguration usually goes after these
+     # and without it many roles do not run
+     extension::FileWatcher
+     support::SignalCatcher
+     support::logging::ConsoleLogHandler
+     support::logging::DataLogger
+     # This must come first as it emits config which many roles depend on (e.g. SensorFilter, WalkEngine)
+     motion::KinematicsConfiguration
+     support::configuration::NetworkConfiguration
+     support::configuration::GlobalConfig
+     platform::darwin::HardwareIO
+     platform::darwin::SensorFilter
+     input::Camera
+     support::NUsight
+     output::ImageCompressor
+     support::configuration::SoccerConfig
+   )
+   ```
+
+2. Build using `./b build` and then install onto a robot using `./b install n<number>`. More on building can be found on the [Getting Started page](/guides/nubots/getting-started) or the [Build System page](/system/nubots/build-system).
+
+3. On the robot, enable compressed image recording in `/config/DataLogging.yaml`. Do this by setting `message.output.CompressedImage` to `true`.
+
+4. Ensure the Left and Right camera serial numbers are correct in their respective configuration files (`Left.yaml` and `Right.yaml`).
+
+   The serial numbers can be retrieved by running the `sudo lsusb -s <bus number>:<device number> -v` command on the robot.
```

Comment:
How are people supposed to know the bus and device number?
if you just run `lsusb` it should show them, and you should pick the ones from the FLIR/Point Grey devices depending on how they show up
Also, you shouldn't need sudo for this command.